### PR TITLE
WaveRubber: a simple waveform analyzer

### PR DESCRIPTION
**Pull request by guprobr**
This kinda elastic analyzer uses no transformations and is colored by a gradient based on the user QT "Highlight" palette.

### CODE REVIEW

File: src/analyzer/waverubber.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  
+  int mid_y = height() / 4;
+  int num_samples = s.size();
+  
+  float x_scale = static_cast<float>(width()) / num_samples;
+  float prev_y = mid_y;
+
+  // Draw the waveform
+  for (int i = 0; i < num_samples; ++i) {
+    	// Normalize amplitude to 0-1 range
```
Comment (jonaski) [inline src/analyzer/waverubber.cpp:67]: Indent should be 2 spaces and no tabs, you can use clang-format to automatically format this, but don't run it on all sources as some things with clang-format formatting is wrong.:

```
sed -i 's/DisableFormat:   true/DisableFormat:   false/g' .clang-format 
clang-format -i src/analyzer/waverubber.*
```

File: src/analyzer/waverubber.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+WaveRubber::WaveRubber(QWidget *parent)
+  : AnalyzerBase(parent, 9) {}
+
+void WaveRubber::resizeEvent(QResizeEvent *e) {
+  Q_UNUSED(e)
+  canvas_ = QPixmap(size());
+  canvas_.fill(palette().color(QPalette::AlternateBase));
+}
+
+void WaveRubber::analyze(QPainter &p, const Scope &s, bool new_frame) {
```
Comment (jonaski) [inline src/analyzer/waverubber.cpp:41]: Add a newline after `{` in every function with more than 1-3 lines.

File: src/analyzer/waverubber.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    	prev_y = y;
+
+  }
+
+  canvas_painter.end();
+  p.drawPixmap(0, 0, canvas_);
+}
+
+void WaveRubber::transform(Scope &s) {
+	// no need transformation for waveform analyzer
```
Comment (jonaski) [inline src/analyzer/waverubber.cpp:88]: Uppercase first letter in comment for consistency.

**Review by jonaski — CHANGES_REQUESTED** [review]
Thanks. Could you fix the formatting and please squash the new commit with the current.

**Comment by guprobr** [comment]
okey changes made ! :D

**Comment by jonaski** [comment]
And please squash the correction commit with the first.

File: src/analyzer/waverubber.cpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  // Clear the canvas
+  canvas_ = QPixmap(size());
+  canvas_.fill(palette().color(QPalette::Window));
+
+  QPainter canvas_painter(&canvas_);
+
+  // Set the pen color to the QT palette highlight color
+  canvas_painter.setPen(palette().color(QPalette::Highlight));
+  // Get pointer to amplitude data 
+  const float* amplitude_data = s.data();
```
Comment (jonaski) [inline src/analyzer/waverubber.cpp:53]:
```suggestion
  const float *amplitude_data = s.data();
```

**Review by jonaski — CHANGES_REQUESTED** [review]

**Comment by jonaski** [comment]
I tested it, looks really nice :)

**Comment by jonaski** [comment]
You can squash with:
`git rebase -i 536d706f007f77fd4ea58d4fe7f645f972f60501~`
Enter `s` in front of the last 2 commits. And push with:
`git push origin WaveRubber_analyzer --force-with-lease`

**Review by jonaski — APPROVED** [review]

**Pull request merged**